receiver/hostmetricsreceiver/internal/scraper.go

```diff
@@ -46,29 +48,13 @@ type Factory interface {
 	CreateMetricsScraper(
 		ctx context.Context,
 		logger *zap.Logger,
-		cfg Config,
-		consumer consumer.MetricsConsumer) (Scraper, error)
+		cfg Config) (Scraper, error)
 }
 
 // Config is the configuration of a scraper.
 type Config interface {
```

Comment:
No need to have this interface you can simply embed the ConfigSettings and load it using viper. No need to expose these getter and setter.

Reply:
The CollectionInterval getter/setters were needed before when the receiver was performing some action related to Collection Intervals: setting the default value or generating tickers.

Now that all of that is gone, it only remains as a type to be returned by the `factory.GetDefaultConfig()` function above. I could change that function definition to just return an `interface{}`, but I think this is tidier (I may be missing something though?)

Reply:
Why do you not return a `ConfigSetting`?

Reply:
Not very important now. We can address this later.

Reply:
I'm fairly new to Go so I could be missing something about how typing works, but how can I return, for example, the [cpuscraper.Config](https://github.com/open-telemetry/opentelemetry-collector/blob/master/receiver/hostmetricsreceiver/internal/scraper/cpuscraper/config.go#L20) as a `ConfigSetting`? It embeds the `ConfigSettings` struct, but is a different type.

Side note: I know most of the scraper configs (other than cpu) are empty right now but most of them won't be in the end (e.g. I need to allow procPath to be configured, and add device filters to the disk/filesystem scrapers, etc)

Reply:
You are correct. Nothing to do here.